### PR TITLE
Rename `pattern` to `design` command in `mesheryctl`

### LINKED ISSUE #10592: [mesheryctl] Rename `pattern` to `design` command
_See [mesheryctl Command Tracker](https://bit.ly/3dqXy1q) for current status of commands._

### Current Behavior

Currently, we have `mesheryctl pattern` command to import, list patterns into Meshery. The objective of the issue is to rename `mesheryctl pattern` to `mesheryctl design`. Subcommands present under `pattern` would be the same, only the name would change

### Implementation

N/A. Only name would change

### Acceptance Tests

Existing tests to be migrated to new name to avoid test failures.

### Mockups

---

### Contributor Resources

- ⌨️ [mesheryctl Contributing Guide](https://github.com/meshery/meshery/blob/master/mesheryctl/README.md)
- [Beginner's guide to contributing to Meshery and mesheryctl](https://youtu.be/hh_kFLZx3G4)
- [mesheryctl Command Tracker](https://docs.google.com/spreadsheets/d/1q63sIGAuCnIeDs8PeM-0BAkNj8BBgPUXhLbe1Y-318o/edit#gid=0)
- [Meshery CLI Commands and Documentation](https://docs.google.com/document/d/1xRlFpElRmybJ3WacgPKXgCSiQ2poJl3iCCV1dAalf0k/edit#heading=h.5fucij4hc5wt)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](http://discuss.meshery.io) and [Layer5 Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by alphaX86**
**Notes for Reviewers**

This PR fixes #10592 

~PR in progress! Please don't merge!~ Ready for review!

Checklist:
- [x] Basic name change in all files and test files
- [x] Test design command test files 
	- [x] Unit tests 
	- [x] Test in real time 
		- [x] apply
		- [x] delete
		- [x] list
		- [x] view
		- [x] onboard
		- [x] offboard
- [x] Verify if results are same as `pattern` in both tests as well as real usage (in progress)
- [x] Complete rename of command
- [x] Update docs
	- [ ] Update index entry (TBD for other commands)
	- [ ] Generate doc pages for dry run

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Syncing with master.

**Comment by MUzairS15** [comment]
@alphaX86 are the checklist in the PR description up-to-date ? Have you performed the testing? // @RipulHandoo

**Comment by leecalcote** [comment]
Current status:

Screenshot 2024-04-11 at 12 02 31 AM

**Comment by leecalcote** [comment]
@lekaf974, you might like to offer review on this PR.

**Comment by l5io** [comment]
🚀 Preview for commit da949ee98115884b9b3c5ce9d4768eff97d794f8 at: https://66177fc7d2eb21462f381180--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 8edfa1c5a35ae8b5eadcba9651c09e82bff9bf98 at: https://66178e53a5c6bf51632114c5--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit f22186a5b511bf03a443296d5bbe3749d18d6d7a at: https://661792bbe2505556cdf25ffd--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 7209cd7ad1fbb0fe48a478b5a412889063de16cf at: https://6617bb7bd2eb21799437ea93--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit a3be89d755c59a523d5f27e0673c4c0e467d18e9 at: https://661bc0b99f80c443ac3260d7--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 83e9f87dd49ff6b5240b77a3e1c7def441c9b22c at: https://661becf6740ddc7ef1de1a19--meshery-docs-preview.netlify.app

**Review by leecalcote — COMMENTED** [review]
Good stuff here, @alphaX86. Will you confirm that each of the GitHub workflows (https://github.com/meshery/meshery/tree/master/.github) that reference `mesheryctl pattern` have been updated as well? 

Please consider this GitHub Action, too:

1. https://github.com/layer5io/meshery-smp-action
2. https://github.com/layer5labs/meshmap-snapshot

**Comment by leecalcote** [comment]
Great checklist, @alphaX86

**Comment by lekaf974** [comment]
> @lekaf974, you might like to offer review on this PR.

Sure taking a look

**Comment by alphaX86** [comment]
Resyncing PR...

File: mesheryctl/internal/cli/root/design/testdata/offboard.output.golden [hunk 1]
```diff
@@ -1,3 +1,3 @@
 pattern file converted to pattern file
```
Comment (lekaf974) [inline mesheryctl/internal/cli/root/design/testdata/offboard.output.golden:1]: does pattern ok here or need to be rename to design ?

**Review by lekaf974 — COMMENTED** [review]
I took time to check and LGTM

**Comment by leecalcote** [comment]
@alphaX86 how are we doing on this diligence? --> https://github.com/meshery/meshery/pull/10603#pullrequestreview-2004157133

**Comment by alphaX86** [comment]
PR staled. Reopening a new one with synced changes

**Pull request closed without merging**